### PR TITLE
Added the control to use self closing tag in XML module and added two unit cases to test the same.

### PR DESCRIPTION
**Pull request by madhavgupta2011**
Added the control to use self closing tag in XML module and added two unit cases to test the same.

### CODE REVIEW

**Comment by stleary** [comment]
* Tests are failing, please fix. 
* The default behavior should not change. Instead, try adding a property to XMLParserConfiguration to allow the developer to change the empty tag behavior. This object is passed to the private XML toString() method, so you don't need to add another parameter.

**Comment by madhavgupta2011** [comment]
Sure. Will do it. Thanks @stleary

File: src/main/java/org/json/XML.java [hunk 1]
```diff
@@ -853,8 +870,15 @@ private static String toString(final Object object, final String tagName, final
                     sb.append(indent(indent));
                     sb.append('<');
                     sb.append(key);
-                    sb.append("/>");
-                    if(indentFactor > 0){
+                    if (!keepSelfClosingTags) {
```
Comment (javadev) [inline src/main/java/org/json/XML.java:873]: `if (!condition) code1 else code2`

can be replaces with

`if (condition) code2 else code1`

**Comment by javadev** [comment]
```
[INFO] Finished at: 2023-07-11T12:12:16Z
[INFO] ------------------------------------------------------------------------
Error:  Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.3.2:testCompile (default-testCompile) on project json: Compilation failure
Error:  /home/runner/work/JSON-java/JSON-java/src/test/java/org/json/junit/XMLTest.java:[1325,33] error: no suitable method found for toString(JSONObject,boolean)
```

File: src/main/java/org/json/XML.java (hunk 1 above)
Comment (madhavgupta2011) [reply src/main/java/org/json/XML.java:873]: Yeah, sure. I'll change it. I am thinking to change my approach though.

**Comment by javadev** [comment]
Since July 24, there have been no changes. Therefore, I believe it would be appropriate to consider canceling the pull request.

**Comment by stleary** [comment]
@madhavgupta2011 Are you able to make the recommended corrections for this PR?
Do you think there is still a need for this change?

**Comment by stleary** [comment]
Thanks, but not accepting this PR. The issue has been reopened.

**Pull request closed without merging**